kf/nfconfig.go

```diff
@@ -942,3 +943,297 @@ func getHostInterfaces() (map[string]bool, error) {
 	}
 	return hostIfaces, nil
 }
+
+func (c *NFConfigs) AddAndStartBPF(bpfProg *models.BPFProgram, ifaceName string, direction string) error {
+	var bpfList *list.List
+	if bpfProg == nil {
+		return fmt.Errorf("AddAndStartBPF - bpf program is nil")
+	}
+
+	if bpfProg.AdminStatus == models.Disabled {
+		return nil
+	}
+
+	bpf := NewBpfProgram(c.ctx, *bpfProg, c.HostConfig)
+
+	switch direction {
+	case models.XDPIngressType:
+		bpfList = c.IngressXDPBpfs[ifaceName]
+	case models.IngressType:
+		bpfList = c.IngressTCBpfs[ifaceName]
+	case models.EgressType:
+		bpfList = c.EgressTCBpfs[ifaceName]
+	default:
+		return fmt.Errorf("unknown direction type")
+	}
+
+	if bpfList == nil {
+		log.Warn().Msgf("%s program list is empty", direction)
+		return nil
+	}
+
+	for e := bpfList.Front(); e != nil; e = e.Next() {
+		data := e.Value.(*BPF)
+		if data.Program.Name == bpfProg.Name {
+			log.Warn().Msgf("%v is already running on %v iface and in %v direction ", data.Program.Name, ifaceName, direction)
+			return nil
+		}
+
+		if data.Program.SeqID == bpfProg.SeqID {
+			log.Warn().Msgf("duplicate seq Id detected for %v in direction %v", data.Program.Name, direction)
+			return nil
+		}
+	}
+	for e := bpfList.Front(); e != nil; e = e.Next() {
+		data := e.Value.(*BPF)
+		if data.Program.SeqID > bpfProg.SeqID {
+			tmpBPF := bpfList.InsertBefore(bpf, e)
+			if err := c.DownloadAndStartBPFProgram(tmpBPF, ifaceName, direction); err != nil {
+				return fmt.Errorf("failed to download and start network function %s version %s iface %s direction %s", bpfProg.Name, bpfProg.Version, ifaceName, direction)
+			}
+
+			if tmpBPF.Next() != nil {
+				if err := c.LinkBPFPrograms(tmpBPF.Value.(*BPF), tmpBPF.Next().Value.(*BPF)); err != nil {
+					log.Error().Err(err).Msg("AddAndStartBPF - failed LinkBPFPrograms after InsertBefore element to with next prog")
+					return fmt.Errorf("AddAndStartBPFProg - failed LinkBPFPrograms after InsertBefore element to with next prog %v", err)
+				}
+			}
+			return nil
+		}
+	}
+
+	// insert at the end
+	if err := c.PushBackAndStartBPF(bpfProg, ifaceName, direction); err != nil {
+		return fmt.Errorf("failed to push back and start network function %s version %s iface %s direction %s", bpfProg.Name, bpfProg.Version, ifaceName, direction)
```

Comment:
```suggestion
		return fmt.Errorf("failed to push back and start eBPF program %s version %s iface %s direction %s", bpfProg.Name, bpfProg.Version, ifaceName, direction)
```
also look for any other occurrences in this file